packages/blocks/src/specs/utils/spec-provider.ts

```diff
@@ -28,6 +28,15 @@ export class SpecProvider {
     this.specMap.delete(id);
   }
 
+  extendSpec(id: string, newSpec: BlockSpec[]) {
+    const existingSpec = this.specMap.get(id);
+    if (!existingSpec) {
```

Comment:
I think we should do nothing but log an error if the `id` to extend is not exists